illumio_connector.py

```diff
@@ -1,249 +1,341 @@
+                phantom.APP_ERROR, "Please enter a valid value for 'port' parameter"
             )
 
-        # Please specify the status codes here
-        if 200 <= r.status_code < 399:
-            return RetVal(phantom.APP_SUCCESS, resp_json)
+        protocol = param["protocol"].lower()
+        if protocol not in ["tcp", "udp"]:
+            return action_result.set_status(
+                phantom.APP_ERROR, "Please enter a valid value for 'protocol' parameter"
+            )
 
-        # You should process the error returned in the json
-        message = "Error from server. Status Code: {0} Data from server: {1}".format(
-            r.status_code,
-            r.text.replace(u'{', '{{').replace(u'}', '}}')
-        )
+        policy_decisions_string = param["policy_decisions"].lower()
 
-        return RetVal(action_result.set_status(phantom.APP_ERROR, message), None)
+        pce = self.connect_pce()
+        if not pce:
+            return action_result.set_status(
+                phantom.APP_ERROR, "Failed to connect to PCE"
+            )
 
-    def _process_response(self, r, action_result):
-        # store the r_text in debug data, it will get dumped in the logs if the action fails
-        if hasattr(action_result, 'add_debug_data'):
-            action_result.add_debug_data({'r_status_code': r.status_code})
-            action_result.add_debug_data({'r_text': r.text})
-            action_result.add_debug_data({'r_headers': r.headers})
+        try:
+            policy_decisions_list = [
+                x.strip() for x in policy_decisions_string.split(",")
+            ]
+            policy_decisions_list = list(filter(None, policy_decisions_list))
 
-        # Process each 'Content-Type' of response separately
+            traffic_query = illumio.TrafficQuery.build(
+                start_date=start_time.isoformat(),
+                end_date=end_time.isoformat(),
+                include_services=[{"port": port, "proto": protocol}],
+                policy_decisions=policy_decisions_list,
+            )
 
-        # Process a json response
-        if 'json' in r.headers.get('Content-Type', ''):
-            return self._process_json_response(r, action_result)
+            traffic_flow_list = pce.get_traffic_flows_async(
+                query_name="phantom_block_port_traffic_query",
+                traffic_query=traffic_query,
+            )
 
-        # Process an HTML response, Do this no matter what the api talks.
-        # There is a high chance of a PROXY in between phantom and the rest of
-        # world, in case of errors, PROXY's return HTML, this function parses
-        # the error and adds it to the action_result.
-        if 'html' in r.headers.get('Content-Type', ''):
-            return self._process_html_response(r, action_result)
+        except IllumioException as e:
+            return action_result.set_status(
+                phantom.APP_ERROR, f"Encountered error while running traffic query: {e}"
```

Comment:
f strings are python 3.7+, do we need to support 3.6?

Reply:
Used .format() instead of f-string